### PR TITLE
Js/rw 594

### PR DESCRIPTION
**Pull request by johnstill**
Changes:

`parameterId`s for search group parameters of type `DEMO AGE` are now cast to string with a prefix rather than remaining raw integers (apparently the hashing algo allowed negative integral outputs which apparently caused some issues, ~although I'm not 100% how~)

Edit: here's how.  Somewhere in the lookup procedure immutable cast the maps to lists of tuples (I think) and the negative integer was interpreted as an offset from the tail of the list rather than as a key in a map by `getIn`; and of course, there isn't an IndexError in js, its just `undefined`.

Similarly, when repopulating the demographics form we now rely on a selection's `parameterId` rather than a hash of the parameter.  This should be faster, but more importantly, in the transition from criteria to parameter the shape of the data changes in some ways, so a parameter loaded via cloning and a criteria loaded via the API would not have the same hash.

### CODE REVIEW

File: ui/src/app/cohort-search/demographics/demographics.component.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ export class DemographicsComponent implements OnInit, OnDestroy {
           operator: 'between',
           operands: [lo, hi],
         });
-        const paramId = attr.hashCode();
+        const paramId = `age-param${attr.hashCode()}`;
         return this.ageNode
```
Comment (calbach) [inline ui/src/app/cohort-search/demographics/demographics.component.ts]: Trying to understand this code: what is the type of `ageNode`?

File: ui/src/app/cohort-search/multi-select/multi-select.component.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export class MultiSelectComponent implements OnInit, OnDestroy {
   @Input() includeSearchBox = true;
   @Input() options = List();
   @Input() set initialSelection(opts) {
-    const _selections = opts.map(opt => opt.hashCode()).toSet();
+    const _selections = opts.map(opt => opt.get('parameterId')).toSet();
```
Comment (calbach) [inline ui/src/app/cohort-search/multi-select/multi-select.component.ts]: Regression test? This seems like a subtle bug and though I'm still trying to understand the code better I'm not sure I would have caught this in code review.

File: ui/src/app/cohort-search/demographics/demographics.component.ts [hunk 3]
```diff
@@ -223,7 +223,7 @@ export class DemographicsComponent implements OnInit, OnDestroy {
...
           .set('parameterId', paramId)
```
Comment (calbach) [inline ui/src/app/cohort-search/demographics/demographics.component.ts]: Not sure I follow - where is this being used as a map key?

File: ui/src/app/cohort-search/multi-select/multi-select.component.ts (hunk 2 above)
Comment (johnstill) [reply ui/src/app/cohort-search/multi-select/multi-select.component.ts]: Tests are added, at least for this aspect.

**Comment by johnstill** [comment]
Did a code walk through offline with CH.  Verbal OK from freeman. Merging.

**Pull request merged**